spinalcordtoolbox/reports/qc2.py

```diff
@@ -150,6 +151,777 @@ def create_qc_entry(
+        """
+        # Rendering options for all the labels.
+        text_args = dict(
+            color='yellow',
+            fontsize=4,
+            path_effects=[
+                mpl_patheffects.Stroke(linewidth=1, foreground='black'),
+                mpl_patheffects.Normal(),
+            ],
+        )
+        for tile_num, (slice_label, coords) in enumerate(self.tiles.items()):
+            # Get real coordinates from the `slice` objects.
+            top, bottom, _ = coords[0].indices(self.canvas.shape[0])
+            left, right, _ = coords[1].indices(self.canvas.shape[1])
+            if tile_num == 0:
+                # Axis labels for the first tile.
+                ((top_label, bottom_label), (left_label, right_label)) = self.axis_labels
+                self.ax.text((left + right) / 2, top, top_label, ha='center', va='top', **text_args)
+                self.ax.text((left + right) / 2, bottom, bottom_label, ha='center', va='bottom', **text_args)
+                self.ax.text(left, (top + bottom) / 2, left_label, ha='left', va='center', **text_args)
+                self.ax.text(right, (top + bottom) / 2, right_label, ha='right', va='center', **text_args)
+            else:
+                # Slice labels for the other tiles.
+                self.ax.text(left, top, slice_label, ha='left', va='top', **text_args)
+
+    def save(self, path: Path):
+        """Save the final figure."""
+        self.fig.savefig(str(path), format='png', transparent=True)
+
+
+def sct_register(
+    fname_input: str,
+    fname_output: str,
+    fname_seg: str,
+    command: str,
+    argv: Sequence[str],
+    path_qc: str,
+    dataset: str | None,
+    subject: str | None,
+    p_resample: float | None = 0.6,
+):
+    """
+    Generate a QC report for sct_register_multimodal or sct_register_to_template.
+
+    Axial orientation, switch between input and output images.
+    """
+    cmdline = [command]
+    cmdline.extend(argv)
+
+    with create_qc_entry(
+        path_input=Path(fname_input).resolve(),
+        path_qc=Path(path_qc),
+        command=command,
+        cmdline=list2cmdline(cmdline),
+        plane='Axial',
+        dataset=dataset,
+        subject=subject,
+    ) as imgs_to_generate:
+
+        img_input = Image(fname_input)
+        img_output = Image(fname_output)
+        img_seg = Image(fname_seg)
+
+        slicing_spec = SlicingSpec.full_axial(img_input, p_resample).center_patches(img_seg)
+
+        for img, path in [
+            (img_input, imgs_to_generate['path_background_img']),
+            (img_output, imgs_to_generate['path_overlay_img']),
+        ]:
+            mosaic = slicing_spec.get_mosaic()
+            mosaic.insert_slices(slicing_spec.get_slices(img, order=2))
+            mosaic.canvas = equalize_histogram(mosaic.canvas)
+            mosaic.ax.imshow(mosaic.canvas, cmap='gray', interpolation='none')
+            mosaic.add_labels()
+            mosaic.save(path)
+
+
+def sct_fmri_compute_tsnr(
```

Comment:
Ditto for the tsnr report!

<img width="1093" height="552" alt="Image" src="https://github.com/user-attachments/assets/00beb717-3d48-4944-8f75-b8fd3acb5898" />

<img width="1093" height="552" alt="Image" src="https://github.com/user-attachments/assets/1193946b-0df8-411b-86bd-efe34aa3f3c1" />